src/UseCase/BannerSelection/BannerSelectionUseCase.php

```diff
@@ -31,35 +31,40 @@ public function __construct( CampaignCollection $campaignCollection, ImpressionT
 	}
 
 	public function selectBanner( Visitor $visitor ): BannerSelectionData {
-		if ( $this->getCurrentCampaign() === null ||
-			$this->impressionThreshold->isThresholdReached( $visitor->getTotalImpressionCount() ) ||
-			$visitor->inCategory( $this->getCurrentCampaign()->getCategory() ) ||
-			$this->isRatioLimited( $this->getCurrentCampaign()->getDisplayPercentage() ) ) {
+		$remainingCampaigns = $this->campaignCollection->filter(
```

Comment:
I like the changes to the campaignCollection a lot. What's going on in the selectBanner method is much clearer now.